### PR TITLE
Issue #74 data attributes added for anchor tag in main section

### LINKED ISSUE #74: Add `data-*` attributes for analytics
Related discussion https://adobe-dx-support.slack.com/archives/C03GY8A76KT/p1677236975034569
More details awaited on this from Nedbank.

### PR DESCRIPTION
**Pull request by jindaliiita**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #74 

Test URLs:
- Before: https://main--nedbank--hlxsites.hlx.live/content/nedbank/za/en/personal/home
- After: https://data-analytics--nedbank--hlxsites.hlx.live/drafts/index-piyush

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -572,6 +580,69 @@ function decorateTemplateAndTheme() {
...
+  }
+
+  return recurAnchor(ele.parentElement, result);
+}
+
+/**
+ * Adding attributes in achor tag for main section
+ * @param {*} a
+ */
+function setAttributesMain(a) {
```
Comment (sdmcraft) [inline scripts/scripts.js:614]: Could you please check?

In https://data-analytics--nedbank--hlxsites.hlx.page/content/nedbank/za/en/personal/home
`data-pagequicklinklocation="Switch to Nedbank, hassle-free | Need help? | main content`

while in https://personal.nedbank.co.za/home.html, it is
`data-pagequicklinklocation="Switch to Nedbank,  hassle-free | main content"`

Comment (jindaliiita) [reply scripts/scripts.js:614]: @sdmcraft According to me the our implementation is correct. As they have included section heading at various places like
<a href="https://personal.nedbank.co.za/promotions/cash-back.html?cmpid=DIS_4sbmL" title="Learn More" data-pagequicklinktype="card" data-pagequicklinkname="Learn More" data-pagequicklinklocation="Summer Prepaid Competition | Promotions | main content">Learn More</a>

Promotions is section heading in this case, so i have applied same logic to Need Help section also.

File: scripts/scripts.js [hunk 2]
```diff
@@ -205,6 +205,14 @@ export function decorateIcons(element) {
   replaceIcons(element);
 
   element.querySelectorAll('span.icon').forEach((span) => {
+    if (span.classList && span.classList.contains('icon-arrow')
```
[the remark this answers is not in the document]
Comment (jindaliiita) [reply scripts/scripts.js:208]: Can be added but currently all the icon-diagonal-arrow have anchor tag as sibling.

[the remark this answers is not in the document]
Comment (sdmcraft) [reply scripts/scripts.js:208]: Ah..I see that the arrows aren't clickable in our implementation while it is on the main site. They need to be clickable. If I make them links in the word doc, they seem to get wrapped in anchor tag though the styling needs to be fixed. Subsequently, these anchor tags should get decorated as usual.

File: scripts/scripts.js [hunk 3]
```diff
@@ -574,6 +574,104 @@ function decorateTemplateAndTheme() {
...
+
+  if (ele.classList.contains('section')) {
+    return result;
+  }
+
+  return recurAnchor(ele.parentElement, result);
+}
+
+/**
+ * Adding attributes in achor tag for main section
```
Comment (sdmcraft) [inline scripts/scripts.js:607]:
```suggestion
 * Adding attributes in anchor tag for main section
```

File: scripts/scripts.js [hunk 4]
```diff
@@ -574,6 +574,104 @@ function decorateTemplateAndTheme() {
...
+}
+
+/**
+ * Decorate anchor tag after all other decorations in order to add data analytics attributes
+ * @param {*} element
+ */
+function decorateAnchor(element) {
+  element.querySelectorAll('a').forEach((a) => {
+    fetchDataAttributesAnchor(a);
+    setAnchorAttributes(a);
```
Comment (sdmcraft) [inline scripts/scripts.js:671]: Should we reverse the order i.e. first set the computed ones and then override if user has specified something in the link?

File: scripts/scripts.js [hunk 5]
```diff
@@ -574,6 +574,104 @@ function decorateTemplateAndTheme() {
...
+
+/**
+ * This function fetches data attributes which were overrided in doc and then setting those.
+ * @param {*} a
+ */
+export function fetchDataAttributesAnchor(a) {
+  const HASH_SEPARATOR = '#';
+  const EQUAL_SEPARATOR = '=';
+
+  if (a.href && a.href.includes(HASH_SEPARATOR)) {
```
Comment (sdmcraft) [inline scripts/scripts.js:645]:
```suggestion
  if (a.href && a.href.includes(HASH_SEPARATOR && a.href.includes("data-")) {
```

File: scripts/scripts.js (hunk 4 above)
Comment (jindaliiita) [reply scripts/scripts.js]: Yes makes sense, redundant if checks removed in this case

**Review by sdmcraft — APPROVED** [review]

**Pull request merged**